Include shared library of non owner users

timvahlbrock opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/steilerDev/icloud-photos-sync/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:


## What is the current behavior?
If the user isn't the owner of the shared zone, photos in the shared zone aren't discovered.

Issue Number: #598


## What is the new behavior?
Photos in the shared zone are discovered, regardless of wether the user is owner of the shared zone or not.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

timvahlbrock commented:
hey @steilerDev, I totally understand that you're probably still busy and wouldn't have time for a review of this PR or the other one. Could you still enable checks on this PR so I can get it into a state where it is ready to review once you have time for it?

steilerDev commented:
@timvahlbrock can you update your PR so I can get it bundled in the next release?

timvahlbrock commented:
FYI: I think I haven't written any tests to test the bug/new functionality yet.

timvahlbrock commented:
@steilerDev I just added some tests. Probably some areas will still need some refactoring, but it would be nice if you could already take a look at the way I implemented this and if you're good with the general idea.

timvahlbrock commented:
Great! If you have some specific things on your mind feel free to post them here, maybe I can take a shot at them myself. I might have some time tomorrow to try to find some possible improvements myself.

timvahlbrock commented:
>especially I want to make sure we have strong typing and I'm not sure I'm in love with the zoneReference type.

I've been thinking about this a bit and patching API objects does feel a bit nasty. But the only two alternatives are

1. Adding an own type that is essentially the API type but with the added `area` field (basically the written out form or the ZoneReference type), or something similar (like the explicit url, an `isOwner` field, ...) and doing a more explicit mapping, which would feel a little overkill, as we're only adding one field.
2. Adding such similar field on the shared zone area  to the resource manager, which would be next to `sharedZone`. This also feels kind of odd as then there would be two fields containing information on the shared zone.
3. Storing the shared zone area within iCloudPhotos, which would probably be the dirtiest solution, as iCloudPhotos as a sort of "API Gateway" shouldn't hold any data itself.

Have you got any ideas on this?

timvahlbrock commented:
Awesome! I should be able to verify this tomorrow.

timvahlbrock commented:
@steilerDev this seems to work fine. I didn't run through the whole synchronisation, but ICPS seems to find the shared library and assets just fine. ✨

timvahlbrock commented:
I don't know if you want those commits to be squashed, so I'm just gonna leave them like they are for now.

steilerDev commented:
Great - I‘ll merge and release this weekend 👍
